expr/types.go

```diff
@@ -80,6 +82,17 @@ type argString struct {
 func (a argString) Key() string    { return a.key }
 func (a argString) Optional() bool { return a.opt }
 
+// like string, but should result in a regex
+type argRegex struct {
+	key       string
+	opt       bool
+	validator []validator
+	val       **regexp.Regexp
```

Comment:
That's also what I thought, but I couldn't get it to work. the problem is the field in the Func is a pointer to a regexp (not strictly needed I think, but otherwise `go vet` complains), and from here we need to modify that value.